app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -28,4 +30,14 @@ class SplashViewModel @Inject constructor(private val navigator: SplashNavigator
         super.onCleared()
         downloadData.dispose()
     }
+
+    fun onActivityResult(requestCode: Int, resultCode: Int) {
+        if (requestCode == SIGN_IN_RESULT_CODE) {
+            if (resultCode == Activity.RESULT_OK) {
```

Comment:
It's better to keep Activity related things in Activity. Check `requestCode` and `resultCode` in Activity and call the appropriate function in the viewmodel.